scaletest/autostart/run_internal_test.go

```diff
@@ -0,0 +1,172 @@
+package autostart
+
+import (
+	"io"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/coder/coder/v2/coderd/coderdtest"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/provisioner/echo"
+	"github.com/coder/coder/v2/provisionersdk/proto"
+	"github.com/coder/coder/v2/scaletest/createusers"
+	"github.com/coder/coder/v2/scaletest/workspacebuild"
+	"github.com/coder/coder/v2/testutil"
+	"github.com/coder/quartz"
+)
+
+func TestRun(t *testing.T) {
+	t.Parallel()
+	numUsers := 2
+	autoStartDelay := time.Minute
+	expectedLatency := 5 * time.Minute
+
+	mClock := quartz.NewMock(t)
+	mClock.Set(time.Now())
```

Comment:
I generally consider this an anti-pattern for Quartz, because it introduces variability into the test, and opens you up to getting flaked by things like daylight savings.

It's also nice to have each run have the same times so you don't have to do math when reading the logs.

Reply:
Another option would be to mock the sdkclient. You still need `Run` with the real client to satisfy the `Runnable` interface, but you could have an internal method that takes an interface with just the sdk methods you need.

You'll also end up having to mock out the other Runners you depend on, but that's also good hygene to keep this a focused unit test instead of a sprawling integration test.